lib/cc/services/github_pull_requests.rb

```diff
@@ -66,11 +88,64 @@ def github_slug
     @payload.fetch("github_slug")
   end
 
+  def author_username
+    @payload.fetch("author_username")
+  end
+
   def response_includes_repo_scope?(response)
     response.headers["x-oauth-scopes"] && response.headers["x-oauth-scopes"].split(/\s*,\s*/).include?("repo")
   end
 
   def test_status_code
     422
   end
+
+  def welcome_comment_implemented?
+    true
+  end
+
+  def user_url
+    "#{config.base_url}/user"
+  end
+
+  def comments_url
+    "#{config.base_url}/repos/#{github_slug}/issues/#{number}/comments"
+  end
+
+  def able_to_comment?
+    response_includes_repo_scope?(service_get(user_url))
+  end
+
+  INTRODUCTION_TEMPLATE = <<-HEADER.freeze
+Hey, @%s-- Since this is the first PR we've seen from you, here's some things you should know about contributing to %s:
+  HEADER
+
+  def welcome_comment_introduction
+    format INTRODUCTION_TEMPLATE, author_username, github_slug
+  end
+
+  def welcome_comment_body
+    config.welcome_comment_markdown
+  end
+
+  ADMIN_ONLY_FOOTER_TEMPLATE = <<-FOOTER.freeze
+* * *
+Quick note: By default, Code Climate will post the above comment on the *first* PR it sees from each contributor. If you'd like to customize this message or disable this, go [here](%s).
+  FOOTER
+
+  def welcome_comment_footer
+    format ADMIN_ONLY_FOOTER_TEMPLATE, @payload.fetch("pull_request_integration_edit_url")
+  end
+
+  def author_is_site_admin?
```

Comment:
naming nit, but this is really checking for if they can admin the repo on cc.com, right? To me "site admin" makes me think of super admins, e.g. our team. Would `author_can_admin_repo?` be accurate here?